client/app/bundles/course/assessment/question/rubric-playground/AnswerEvaluationsTableHeader.tsx

```diff
@@ -0,0 +1,172 @@
+        dispatch(
+          questionRubricsActions.initializeAnswerEvaluations({
+            answerIds: data.addAnswerIds,
+            rubricId: selectedRubric.id,
+          }),
+        );
+        await initializeAnswerEvaluations(selectedRubric.id, data.addAnswerIds);
+        break;
+      }
+      case AddSampleMode.RANDOM_STUDENT: {
+        const randomAnswerIds = sampleSize(
+          selectableAnswers,
+          data.addRandomAnswerCount,
+        ).map((answer) => answer.id);
+        dispatch(
+          questionRubricsActions.initializeAnswerEvaluations({
+            answerIds: randomAnswerIds,
+            rubricId: selectedRubric.id,
+          }),
+        );
+        await initializeAnswerEvaluations(selectedRubric.id, randomAnswerIds);
+        break;
+      }
+      case AddSampleMode.CUSTOM_ANSWER: {
+        const mockAnswerId = await createQuestionMockAnswer(
+          data.addMockAnswerText,
+        );
+        dispatch(
+          questionRubricsActions.initializeMockAnswer({
+            rubricId: selectedRubric.id,
+            mockAnswerId,
+            answerText: data.addMockAnswerText,
+          }),
+        );
+        await initializeMockAnswerEvaluations(selectedRubric.id, [
+          mockAnswerId,
+        ]);
+        break;
+      }
+      default: {
+        break;
+      }
+    }
+  };
+
+  return (
+    <>
+      <div className="flex flex-row space-x-4 items-center py-2">
+        <Typography variant="h6">Sample Answer Evaluations</Typography>
+        <Button
+          onClick={() => setIsAddingAnswers(true)}
+          size="small"
+          startIcon={<Add />}
+          variant="outlined"
+        >
+          Add Sample Answers
+        </Button>
+        {Boolean(answerCount) && (
+          <Button
+            disabled={answerEvaluationTableData.some((row) => row.isEvaluating)}
+            onClick={() => {
+              const rowsToEvaluate =
+                answerEvaluatedCount === answerCount
+                  ? answerEvaluationTableData
+                  : answerEvaluationTableData.filter(
+                      (answer) => !isAnswerAlreadyEvaluated(answer),
+                    );
+              Promise.all(
+                rowsToEvaluate.map((row) =>
+                  requestRowEvaluation(dispatch, row, selectedRubric.id),
+                ),
+              );
+            }}
+            size="small"
+            startIcon={
+              answerEvaluatedCount === answerCount ? <Refresh /> : <PlayArrow />
+            }
+            variant="outlined"
+          >
+            {!answerEvaluatedCount && `Evaluate All (${answerCount})`}
+            {Boolean(answerEvaluatedCount) &&
+              answerEvaluatedCount === answerCount &&
+              `Re-evaluate All (${answerCount})`}
+            {Boolean(answerEvaluatedCount) &&
+              answerEvaluatedCount !== answerCount &&
+              `Evaluate Remaining (${answerCount - answerEvaluatedCount}`}
```

Comment:
missing a closing bracket ), also should move to translations in the clean-up PR.